src/circularbuffer.jl

```diff
@@ -0,0 +1,104 @@
+
+# this represents a circular buffer, where new items are pushed to the back of the list, 
+# overwriting values in a circular fashion
+# iterating occurs from the "start" until the "end", and will return "length(buffer)" nodes
+
+type CircularBuffer{T} <: AbstractVector{T}
+  capacity::Int
+  startIdx::Int
+  buffer::Vector{T}
+
+  CircularBuffer{T}(::Type{T}, capacity::Int) = new(capacity, 1, T[])
+end
+
+CircularBuffer{T}(::Type{T}, capacity::Int) = CircularBuffer{T}(T, capacity)
+
+function CircularBuffer{T}(::Type{T}, capacity::Int, fillval::T)
+  cb = CircularBuffer(T, capacity)
+  for i in 1:capacity
+    push!(cb, copy(fillval))
+  end
+  cb
+end
```

Comment:
Not sure this constructor is necessary. `fill!(CircularBuffer(T, n), fillval)` ought to work and is more idiomatic. If performance is an issue you could create a specialized method to do this. (Not sure it should be copying `fillval` though.)